src/Hyperledger.Aries.Routing.Edge/EdgeClientService.Backup.cs

```diff
@@ -128,15 +131,31 @@ public async Task<List<Attachment>> RetrieveBackupAsync(IAgentContext context, s
             
             await Task.Run(() => File.WriteAllBytes(tempWalletPath, walletToRestoreInBytes));
             
-            var json = new { path = tempWalletPath, key = seed }.ToJson();
+            var oldAgentOptionsString = JsonConvert.SerializeObject(agentoptions);
 
-            await walletService.DeleteWalletAsync(agentoptions.WalletConfiguration, agentoptions.WalletCredentials);
+            var json = new { path = tempWalletPath, key = seed }.ToJson();
             
-            // Add 1 sec delay to allow filesystem to catch up
-            await Task.Delay(TimeSpan.FromSeconds(1));
+            var newWalletCredKey = Utils.GenerateRandomAsync(32);
+            var newWalletConfigId = Guid.NewGuid().ToString();
+            agentoptions.WalletConfiguration.Id = newWalletConfigId;
```

Comment:
How will a client know that we generated new wallet Id and password? Assigning it to AgentOptions will only persist for the lifetime of DI?